viewing_party/party.py

```diff
@@ -1,23 +1,320 @@
+               
+    return most_frequent_genre
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_friends_watched(user_data):
+    """
+    Helper function to get list of all movies that friends have watched
+    """
+    friends_watched= []
+    
+    for friend in user_data["friends"]:
+        friends_watched.extend(friend["watched"])
         
+    return friends_watched
+
+
+def get_unique_watched(user_data):
+    """
+    This function determines which movies the user has watched 
+    but none of the friends have watched.
+    
+    Parameters:
+        user_data(dict): has two keys:
+            "watched": List of movie dictionaries the user has watched, each with a "title".
+            "friends": List of friends, each having a "watched" list of movie dictionaries.
+
+    Returns:
+        list of dict of movies.
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+    
+    unique_watched = []
+
+    for movie in user_watched:
+        if movie not in friends_watched:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+
+def get_friends_unique_watched(user_data):
+    """
+    This function determines which movies the user has not watched 
+    but at least one of the friends have watched.
+    
+    Parameters:
+        user_data(dict): has two keys:
+            "watched": List of movie dictionaries the user has watched, each with a "title".
+            "friends": List of friends, each having a "watched" list of movie dictionaries.
+
+    Returns:
+        list of dict of unique movies.
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+
+    unique_friends_watched = []
+
+    for movie in friends_watched:
+        if movie not in user_watched and movie not in unique_friends_watched:
+            unique_friends_watched.append(movie)
+    
+    return unique_friends_watched
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+def get_available_recs (user_data):
+    """
+    Return a list of recommended movies based on the user's subscriptions 
+        and their's friends watched movies
+        
+    Parameters:
+        user_data (dict): has two keys:
+            "subscriptions": List of streaming services the user has access to
+            "friends": List of friends, each having a "watched" list of movie dictionaries, 
+                each movie has a "host"
+    
+    Returns:
+        list of dict of movies
+    """
+    
+    user_watched = user_data["watched"]
+    friends_watched = get_friends_watched(user_data)
+  
+    recommended_movies = []
+    for movie in friends_watched:
+        if movie not in user_watched and movie["host"] in user_data["subscriptions"] \

```

Comment:
Remember that get_friends_watched is going to hold all the movies your friends have seen and you haven't! Therefore, we don't actually need to check and make sure there is no overlap. Also, because recommended_movies starts empty, we have other checks that make sure duplicates aren't added. As a result, the first and last conditionals aren't necessary!